### PR TITLE
Upgrade SUMO to 1.12.0

### LINKED ISSUE #1717: Update SUMO to 1.12

### PR DESCRIPTION
**Pull request by saulfield**
Closes #1717

### CODE REVIEW

File: examples/rl/racing/Dockerfile [hunk 1]
```diff
@@ -27,7 +27,7 @@ ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 RUN pip install --upgrade pip wheel
 
 # Setup SUMO.
-RUN pip install eclipse-sumo==1.10.0
+RUN pip install eclipse-sumo==1.12.0
```
Comment (Adaickalavan) [inline examples/rl/racing/Dockerfile:30]: Examples are tested to work with the given dependencies and are defined by their `setup.py`. In general, it is better to leave examples untouched unless they have been tested again.

File: requirements.txt [hunk 2]
```diff
@@ -24,7 +24,7 @@ constantly==15.1.0
 coverage==6.5.0
 cycler==0.11.0
 dm-tree==0.1.7
-eclipse-sumo==1.10.0
+eclipse-sumo==1.12.0
```
Comment (Adaickalavan) [inline requirements.txt:27]: No need to modify the `requirements.txt` file, as it will be automatically overwritten by the CI `ci-auto-commit-linux.yml`.

**Comment by Gamenot** [comment]
Interesting that `test_map` failed https://github.com/huawei-noah/SMARTS/actions/runs/3549894566/jobs/5962761264#step:5:588

I wonder how the new version of SUMO had an effect on it.

**Comment by saulfield** [comment]
> Interesting that `test_map` failed https://github.com/huawei-noah/SMARTS/actions/runs/3549894566/jobs/5962761264#step:5:588
> 
> I wonder how the new version of SUMO had an effect on it.

It looks like it's a small numerical difference:

```
>       assert db == 134.01
E       assert 134.0 == 134.01

smarts/core/tests/test_map.py:135: AssertionError
```

I can change the assert from an equality to `math.isclose()`. It's a bit odd that this passes locally for me though.

File: examples/rl/racing/Dockerfile (hunk 1 above)
Comment (saulfield) [reply examples/rl/racing/Dockerfile]: Ok, I've reverted this.

File: smarts/core/tests/test_map.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ def test_sumo_map(sumo_scenario):
     rpt = RoadMap.Route.RoutePoint(pt=point)
     rendpt = RoadMap.Route.RoutePoint(pt=Point(198, 65.20))
     db = route.distance_between(rpt, rendpt)
-    assert db == 134.01
+    assert math.isclose(db, 134.01)
```
Comment (Gamenot) [inline smarts/core/tests/test_map.py:135]: It might need a bit more tweaking since this still resolves false.

**Review by Gamenot — APPROVED** [review]

**Review by Adaickalavan — APPROVED** [review]

**Comment by Adaickalavan** [comment]
@saulfield 
+ The issue #1155 previously reported for `sumo==1.11` is also encountered in `sumo==1.12`.
+ This might need to be fixed before upgrading sumo beyond version `1.10.0`.

**Pull request merged**